OneSignalSDK/onesignal/src/main/java/com/onesignal/OutcomeEventsController.java

```diff
@@ -31,26 +37,46 @@
     @Nullable
     private OneSignal.OutcomeSettings outcomeSettings;
 
-    OutcomeEventsController(@NonNull OSSessionManager osSessionManager, @NonNull OutcomeEventsRepository outcomeEventsRepository) {
-        this.outcomeEventsRepository = outcomeEventsRepository;
+    public OutcomeEventsController(@NonNull OSSessionManager osSessionManager, @NonNull OutcomeEventsRepository outcomeEventsRepository) {
         this.osSessionManager = osSessionManager;
+        this.outcomeEventsRepository = outcomeEventsRepository;
+
+        initUniqueOutcomeEventsSentSets();
     }
 
     OutcomeEventsController(@NonNull OSSessionManager osSessionManager, @NonNull OneSignalDbHelper dbHelper, @Nullable OneSignal.OutcomeSettings outcomeSettings) {
         this.outcomeEventsRepository = new OutcomeEventsRepository(dbHelper);
         this.osSessionManager = osSessionManager;
         this.outcomeSettings = outcomeSettings;
+
+        initUniqueOutcomeEventsSentSets();
+    }
+
+    /**
+     * Init the sets used for tracking attributed and unattributed unique outcome events
+     */
+    private void initUniqueOutcomeEventsSentSets() {
+        attributedUniqueOutcomeEventsSentSet = OSUtils.newConcurrentSet();
+        unattributedUniqueOutcomeEventsSentSet = OSUtils.newConcurrentSet();
```

Comment:
if i close and re open on less that 30 seconds in theory unattributedUniqueOutcomeEventsSentSet should still keep the last uniqueOutcomes, but well we can agree on ignoring this case

Reply:
You're right, I will fix this for unattributedUniqueOutcomeEventsSentSet

Reply:
@mikechoch Fixed this.